src/lib.rs

```diff
@@ -2365,6 +2386,25 @@ impl Build {
         Ok(())
     }
 
+    fn add_inherited_rustflags(&self, cmd: &mut Tool, target: &TargetInfo) -> Result<(), Error> {
+        let env_os = match self.getenv("CARGO_ENCODED_RUSTFLAGS") {
+            Some(env) => env,
+            // No encoded RUSTFLAGS -> nothing to do
+            None => return Ok(()),
+        };
+
+        let Tool {
+            family, path, args, ..
+        } = cmd;
+
+        let env = env_os.to_string_lossy();
+        let codegen_flags = RustcCodegenFlags::parse(&env)?;
+        let mut cc_flags_args: Vec<OsString> = vec![];
+        codegen_flags.cc_flags(self, path, *family, target, &mut cc_flags_args);
+        args.extend(cc_flags_args);
```

Comment:
Hmmm is there any lifetime issue you've encountered ?

```suggestion
        codegen_flags.cc_flags(self, path, *family, target, &mut args);
```

Reply:
ah yes this can now be shortened, sure why not